### ISSUE TITLE
ElasticSearch 5.2 support

### ISSUE THREAD

**Issue by gvsrini** [jprante/elasticsearch-jdbc#942]
Just curious to know - does the latest **elasticsearch-jdbc** release version support **ES 5.2**.

**Comment by apatrida** [comment]
@gvsrini looking at other issues here it does not yet.

@jprante has done a more thorough job here, but if you need ES 5 support in the meantime you can try: https://github.com/kohesive/elasticsearch-data-import-handler but don't expect the same feature set, for example CRON would be done outside the process.  @jprante I hope you don't mind me putting this here, these issues are just aging so giving an alternative.  I respect your work, not trying to steal from it :-D

**Comment by jprante** [comment]
The latest release does not support ES 5.2.

Implementation for ES 5.x has begun a few months but is not complete, due to lack of time.

@apatrida do not worry, it's always a good thing to learn about alternatives and help users. This is a spare time project only, in the hope to be useful.

**Comment by apatrida** [comment]
I'd like to merge efforts with a lot of your stuff, also spare time hopefully will get to do that...

**Comment by gvsrini** [comment]
I can join in the development in the hope that we can together make something useful to us as well as the community. 

Please let me know where/how to get started. 

Regards 
Srinivas Gattu 

To: "jprante/elasticsearch-jdbc" <elasticsearch-jdbc@noreply.github.com> 
Cc: "gvsrini" <srinivas@callippus.co.uk>, "Mention" <mention@noreply.github.com> 
Sent: Thursday, 30 March, 2017 04:03:35 
Subject: Re: [jprante/elasticsearch-jdbc] ElasticSearch 5.2 support (#942) 



I'd like to merge efforts with a lot of your stuff, also spare time hopefully will get to do that...

**Comment by gvsrini** [comment]
You solution looks WOW. 

Will give it a good try. Right now, I downgraded by ES to 2.4 just because I desperately needed JDBC Importer. 

Your project seems to be a life saver for me. 

Regards 
Srinivas Gattu

**Comment by sanaulla123** [comment]
@apatrida Does it support creating nested objects and array of nested object just like @jprante 's version of JDBC importer

**Comment by apatrida** [comment]
yes, the readme doc shows an example of this using 'struct' and 'collect_list' and 'collect_set' SQL functions.  if you have further questions ask there.
